### PR TITLE
Add support of proxy environment variable (#191)

### LINKED ISSUE #191: [FEATURE] Add support of corporate proxy
* **What are you trying to do?**
Running auditJs on a CI server that sits behind a corporate firewall. 

The operation team opened the url "https://ossindex.sonatype.org/"  from the CI server to the internet (verified with a curl command) but the auditJs tool hangs when "Submitting coordinates to Sonatype OSS Index"

* **What feature or behavior is this required for?**
Being able to automate the vulnerabilities check in our CI before releasing code.

* **How could we solve this issue? (Not knowing is okay!)**
Take into account an HTTP_PROXY env variable or the npmrc "proxy" config to allow connecting through the corporate proxy

* **Anything else?**
I have seen that issue #26 was about implementing this (reading npmrc "proxy" config) : https://github.com/sonatype-nexus-community/auditjs/commit/6570b7c6753f4eacf4198d73ce91160976391bb3
But at that time you were using Request to make network requests. At some point, you switched to node-fetch and I think this feature got lost in the process.

cc @bhamail / @DarthHater / @allenhsieh / @ken-duck

### PR DESCRIPTION
**Pull request by francois-roget**
Allow the CLI to take into account http_proxy or https_proxy environment varaibles to perform the network requests. This is particularly handy in corporate CI.

This pull request makes the following changes:
* Add a getHttpAgent method in RequestHelpers to get a configured http.Agent (or undefined)
* Use this http.Agent for all fetch calls.
* Test the http.Agent generation in RequestHelpers.spec

It relates to the following issue #s:
* Fixes #191 

cc @bhamail / @DarthHater / @allenhsieh / @ken-duck

### CODE REVIEW

File: src/Services/RequestHelpers.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 import os from 'os';
+import { Agent } from 'http';
+const HttpsProxyAgent = require('https-proxy-agent');
```
Comment (DarthHater) [inline src/Services/RequestHelpers.ts:18]: Hmmm, can this not be imported like `import {HttpsProxyAgent} from 'https-proxy-agent';`?

**Comment by DarthHater** [comment]
@francois-roget one minor question, if you wouldn't mind taking a look!

File: src/Services/RequestHelpers.ts (hunk 1 above)
Comment (francois-roget) [reply src/Services/RequestHelpers.ts]: @DarthHater the code has been changed as requested.
I made a fixup. If this version is agreed, I'll squash it to have a single commit

**Comment by francois-roget** [comment]
@DarthHater I corrected according to your remark.

**Pull request merged**